pi-manage

```diff
@@ -1286,8 +1289,130 @@ def import_tokens(file, tokenrealm=None):
+            of = kwargs.pop('output', None)
+            fmt = kwargs.pop('format', 'python')
+            res = f(*args, **kwargs)
+            # format output according to the format parameter
+            if fmt not in fmt_dict:
+                print('Format {0!s} is not defined, using "python" instead.'.format(fmt))
+                fmt = 'python'
+            res = fmt_dict.get(fmt)(res) + '\n'
+            if of and not of == '-':
+                with open(of, 'w') as outfile:
+                    outfile.write(res)
+            else:
+                sys.stdout.write(res)
+        return exp_wrapper
+
+    # now create the command from the wrapper function
+    cmd = Command(exporter_wrapper(func))
+    # and add all the options
+    for o in opts:
+        cmd.add_option(o)
+    return cmd
+
+
+config_exporter_manager = Manager(help='registered exporter commands',
+                                  description='registered exporters')
+for e, f in EXPORT_FUNCTIONS.items():
+    config_exporter_manager.add_command(e, exporter_command(f))
+
+config_manager.add_command('exporter', config_exporter_manager)
+
+inp_fmt_dict = {'python': ast.literal_eval,
+                'json': json.loads,
+                'yaml': yaml.safe_load}
+
+
+def importer_command(func):
+    """ Return a flask_script Command object for importing data"""
+    opts = []
+    # get parameter of function to call
+    arguments, varargs, keywords, defaults = inspect.getargspec(func)
+    opts.append(Option('-i', '--input',
+                       help='The filename to import the data from,'
+                            ' default is <stdin>'))
+    opts.append(Option('-f', '--format', default='python',
+                       help='input format (*python*, json, yaml)'))
+
+    def importer_wrapper(f):
+        # Add the input routines to the import function
+        @wraps(f)
+        def imp_wrapper(*args, **kwargs):
+            d = None
+            ifname = kwargs.pop('input', None)
+            _fmt = kwargs.pop('format', 'python')
+            if ifname and not ifname == '-':
+                with open(ifname, 'r') as infile:
+                    content = infile.read()
+            else:
+                content = sys.stdin.read()
+            for tp in inp_fmt_dict:
+                try:
+                    print(tp)
+                    d = inp_fmt_dict[tp](content)
+                    break
+                except Exception as e:
+                    print(e)
+                    continue
+            if d:
+                print(d)
+                f(d, *args, **kwargs)
+            else:
+                print('Could not read input!')
+        return imp_wrapper
+
+    # now create the command from the wrapper function
+    cmd = Command(importer_wrapper(func))
+    # and add all the options
+    for o in opts:
+        cmd.add_option(o)
+    return cmd
 
+
+config_importer_manager = Manager(help='registered importer commands',
```

Comment:
see my above comment about the phrase "registered".

Reply:
Done in 426389b